client/state/themes/actions/request-theme.js

```diff
@@ -101,3 +102,82 @@ export function requestTheme( themeId, siteId, locale ) {
 			} );
 	};
 }
+
+export function requestSingleTheme( themeId, siteId, sourceType, locale ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: THEME_REQUEST,
+			siteId,
+			themeId,
+		} );
+		if ( sourceType === 'wporg' ) {
```

Comment:
Maybe we could wrap all these conditions in a switch statement and extract the calls to functions as you did with `makeWpcomRequest`. Right now every if gets checked (except for that fishy return statement)